### PR TITLE
refined logging

### PR DESCRIPTION
**Pull request by basit9958**
Signed-off-by: Basit Hasan <hasanbasit13@gmail.com>

### Related changes

- [x] Need to update `chaos-mesh/website`
- [ ] Need to update `Dashboard UI`
- Need to **cheery-pick to release branches**
  - [ ] release-2.1
  - [ ] release-2.0

### Checklist

Tests

- [ ] Unit test
- [ ] E2E test
- [ ] No code
- [ ] Manual test (add steps below)

Side effects

- [ ] Breaking backward compatibility

### Release note 

```text
Please add a release note.

You can safely ignore this section if you don't think this PR needs a release note.
```

### DCO

If you find the DCO check fails, please run commands like below (Depends on the actual situations. For example, if the failed commit isn't the most recent) to fix it:

```shell
git commit --amend --signoff
git push --force
```

### CODE REVIEW

**Comment by basit9958** [comment]
cc @STRRL

File: pkg/dashboard/store/metrics/metrics.go [hunk 1]
```diff
@@ -43,6 +42,7 @@ type Collector struct {
 // NewCollector initializes metrics and collector
 func NewCollector(experimentStore core.ExperimentStore, scheduleStore core.ScheduleStore, workflowStore core.WorkflowStore) *Collector {
 	return &Collector{
+		log:             NewDefaultZapLogger(),
```
Comment (STRRL) [inline pkg/dashboard/store/metrics/metrics.go:45]: please do not instant new logger inside of the constructor, acquire it as one of constructor parameter.

**Review by STRRL — CHANGES_REQUESTED** [review]
Thanks for your contributrion!

PTAL!

**Comment by basit9958** [comment]
cc @STRRL  review please

File: pkg/dashboard/store/metrics/metrics.go [hunk 2]
```diff
@@ -41,8 +39,9 @@ type Collector struct {
 }
 
 // NewCollector initializes metrics and collector
-func NewCollector(experimentStore core.ExperimentStore, scheduleStore core.ScheduleStore, workflowStore core.WorkflowStore) *Collector {
+func NewCollector(log logr.Logger, experimentStore core.ExperimentStore, scheduleStore core.ScheduleStore, workflowStore core.WorkflowStore) *Collector {
 	return &Collector{
+		log:             NewDefaultZapLogger(),
```
Comment (STRRL) [inline pkg/dashboard/store/metrics/metrics.go:44]: please use `log` in parameter instead of `NewDefaultZapLogger()`

**Review by STRRL — CHANGES_REQUESTED** [review]

**Comment by STRRL** [comment]
Hi @basit9958, do you need some help? :)

**Comment by basit9958** [comment]
> Hi @basit9958, do you need some help? :)

I will resolve this by today 
😸

**Comment by basit9958** [comment]
cc @STRRL

File: pkg/dashboard/store/metrics/metrics.go [hunk 3]
```diff
@@ -18,19 +18,18 @@ package metrics
...
 	"github.com/chaos-mesh/chaos-mesh/pkg/dashboard/core"
 )
 
 const chaosDashboardSubsystem = "chaos_dashboard"
 
-var log = ctrl.Log.WithName("metrics-collector")
-
 // Collector implements prometheus.Collector interface
 type Collector struct {
+	log             logr.Logger
```
Comment (STRRL) [inline pkg/dashboard/store/metrics/metrics.go:32]:
```suggestion
	Log             logr.Logger
```

It should use exported field with `fx` dependency injection.

rest LGTM! You are almost there.

File: pkg/dashboard/store/metrics/metrics.go [hunk 4]
```diff
@@ -146,14 +146,14 @@ func (collector *Collector) collectArchivedWorkflows() {
 
 type Params struct {
 	fx.In
-
+	log             logr.Logger
```
Comment (STRRL) [inline pkg/dashboard/store/metrics/metrics.go:149]:
```suggestion
	Log             logr.Logger
```
It should use exported field with fx dependency injection.

rest LGTM! You are almost there.

**Review by STRRL — APPROVED** [review]
LGTM!

**Review by iguoyr — APPROVED** [review]
LGTM

**Pull request merged**